server/src/main/java/org/opensearch/index/codec/composite/Composite99DocValuesReader.java

```diff
@@ -29,11 +60,123 @@
  */
 @ExperimentalApi
 public class Composite99DocValuesReader extends DocValuesProducer implements CompositeIndexReader {
-    private DocValuesProducer delegate;
+    private static final Logger logger = LogManager.getLogger(Composite99DocValuesReader.class);
+
+    private final DocValuesProducer delegate;
+    private IndexInput dataIn;
+    private ChecksumIndexInput metaIn;
+    private final Map<String, Tree> starTreeMap = new LinkedHashMap<>();
```

Comment:
Usually only metadata is read during the data reader initialization, do we really need to initialize the data part here - why not do it during getCompositeField() ?